### PR TITLE
[wip][CARBONDATA-2740]flat folder handling for implicit column and other bug fix

### PR DESCRIPTION
**Pull request by akashrn5**
### Problem
1) When flat folder is enabled for table, pruning is not happening for implicit column.
2) error message is wrong when complex column which does not have any string datatype column is given in local dictionary and exclude column.
3) when data load is failed, corresponding segment file is not getting deleted.

### Solution:
1) when flat folder is enabled, based on the segment number present in the carbondata file name,  pruning is taken care
2) error message is corrected
3) when data load is failed, segment file will be deleted

### CODE REVIEW

File: integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonRelation.scala [hunk 1]
```diff
@@ -173,15 +172,38 @@ case class CarbonRelation(
...
+              .getDirectorySize(CarbonTablePath.getFactDir(carbonTable.getTablePath))
+            val lockDirSize = FileFactory
+              .getDirectorySize(CarbonTablePath.getLockFilesDirPath(carbonTable.getTablePath))
+            val datamaps = DataMapStoreManager.getInstance().getAllDataMap(carbonTable)
+            var datamapsDirectorySize = 0L
+            if (datamaps.size() > 0) {
+              datamaps.asScala.foreach { datamap =>
+                datamapsDirectorySize = datamapsDirectorySize + FileFactory
+                  .getDirectorySize(
+                    carbonTable.getTablePath + datamap.getDataMapSchema.getDataMapName)
```
Comment (KanakaKumar) [inline integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonRelation.scala]: Please use a utility method to make to dm path like org.apache.carbondata.core.util.path.CarbonTablePath#getDataMapStorePath.

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5825/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5826/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5834/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5837/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7169/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5945/

**Comment by akashrn5** [comment]
please refer #2514 , as some chnages are moved to that PR, and flat folder support for implicit column will be raised in another PR

**Pull request closed without merging**